app/jobs/build_job.rb

```diff
@@ -0,0 +1,53 @@
+require "github_client"
+require "task_runner"
+
+class BuildJob
+  attr_reader :pull_request
+
+  def initialize(pull_request)
+    @pull_request = pull_request
+  end
+
+  def perform
+    client.send_repo_status([full_name, sha, "pending"])
+
+    result = TaskRunner.run_build(pull_request)
+
+    status_info = configure_response(result)
+
+    client.send_repo_status(status_info)
+  end
+
+  private
+
+  def client
+    pull_request_user = User.find_by(uid: pull_request[:uid])
+
+    client = GithubClient.new(
+      pull_request_user,
+      Octokit::Client.new(
+        access_token: pull_request_user.github_access_token,
+      ),
+    )
+
+    client
```

Comment:
why set this to a local variable only to return? just let the GitHubClient instantiation do that.